internal/cmd/start.go

```
@@ -0,0 +1,160 @@
+			ForceConflicts:   forceConflicts,
+			InitiatedMsg:     "start initiated",
+			Namespace:        namespace,
+			NewShutdownValue: false,
+			Mapping:          mapping,
+		}
+
+		cluster, err := getPostgresCluster(client, requestArgs)
+		if err != nil {
+			return err
+		}
+
+		err = patchClusterShutdown(cluster, client, requestArgs)
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+
+	return cmdStart
+}
+
+func patchClusterShutdown(cluster *unstructured.Unstructured, client dynamic.NamespaceableResourceInterface, args ShutdownRequestArgs) error {
+	ctx := context.Background()
+
+	currShutdownVal, found, err := unstructured.NestedBool(cluster.Object, "spec", "shutdown")
+	if err != nil {
+		return err
+	}
+	// If the shutdown status is equal to the intent of the command, do nothing.
+	if found && currShutdownVal == args.NewShutdownValue {
+		fmt.Fprintf(args.Config.Out, args.DoNothingMsg)
+		return nil
+	}
+
+	// Construct the payload.
+	intent := new(unstructured.Unstructured)
+	if err := internal.ExtractFieldsInto(cluster, intent, args.Config.Patch.FieldManager); err != nil {
+		return err
+	}
+	if err := unstructured.SetNestedField(intent.Object, args.NewShutdownValue, "spec", "shutdown"); err != nil {
+		fmt.Fprint(args.Config.Out, err.Error())
+		return err
+	}
+	patch, err := intent.MarshalJSON()
+	if err != nil {
+		return err
+	}
+	patchOptions := metav1.PatchOptions{}
+	if args.ForceConflicts {
+		b := true
+		patchOptions.Force = &b
+	}
+
+	// Patch the update.
+	_, err = client.Namespace(args.Namespace).Patch(ctx,
+		args.ClusterName, types.ApplyPatchType, patch,
+		args.Config.Patch.PatchOptions(patchOptions))
+	if err != nil {
+		if apierrors.IsConflict(err) {
+			fmt.Fprintf(args.Config.Out, "SUGGESTION: The --force-conflicts flag may help in performing this operation.\n")
+		}
+		return err
+	}
+	fmt.Fprintf(args.Config.Out, "%s/%s %s\n",
+		args.Mapping.Resource.Resource, args.ClusterName, args.InitiatedMsg)
+	return nil
```

This function does a lot of printing. What would it look like if we returned `string, error`, and the calling function printed the message if it existed and returned the error?

---

Or really, maybe our 3 states are:
1. error -- in which case we want to bubble the error up (and only in the case of the force-conflicts do we have an additional string to print)
2. success -- print success message
3. donothing -- print the do-nothing message

---

That's a better way to go.